File: test-accesstoken-pkg/ts/test-docs-api-key.ts
/**
 * Mirrors website/public/guides/api-key.md against @imbrace/sdk@1.0.4 (npm)
 * — Access Token auth.
 */
import "dotenv/config"
import { ImbraceClient } from "@imbrace/sdk"

const accessToken    = process.env.IMBRACE_ACCESS_TOKEN
const organizationId = process.env.IMBRACE_ORGANIZATION_ID
const baseUrl        = process.env.IMBRACE_GATEWAY_URL ?? "https://app-gatewayv2.imbrace.co"

if (!accessToken || !organizationId) { console.error("Missing creds"); process.exit(1) }

const client = new ImbraceClient({ accessToken, organizationId, baseUrl, timeout: 10_000 })

let pass = 0, fail = 0, skip = 0
const fails: string[] = []

async function step(label: string, fn: () => Promise<unknown>) {
  process.stdout.write(`  • ${label} ... `)
  try { const r = await fn(); console.log(`✓ ${JSON.stringify(r ?? {}).slice(0, 90)}`); pass++ }
  catch (err: any) { console.log(`✗ [${err?.statusCode ?? err?.message}]`); fail++; fails.push(`${label} → ${err?.statusCode ?? err?.message}`) }
}
function skipped(label: string, reason: string) { console.log(`  - ${label}  ⏭ ${reason}`); skip++ }

console.log(`\n━━━ DOCS: guides/api-key.md — auth: ACCESS TOKEN (npm @imbrace/sdk@1.0.4) ━━━\n`)

console.log("══ §Option 2 — programmatic via SDK (requires access-token, which we have) ══")
await step("client.auth.getThirdPartyToken(30) — access-token client should succeed",
  () => (client.auth as any).getThirdPartyToken(30))

console.log("══ §Using the key — re-construct client (covered by §1 init) ══")
skipped("new ImbraceClient({ apiKey }).platform.getMe()", "covered by test-api-pkg")

console.log(`\n━━━ Summary (api-key / Access Token) ━━━`)
console.log(`pass=${pass}  fail=${fail}  skip=${skip}`)
if (fails.length) { console.log("Failures:"); fails.forEach(f => console.log(`  - ${f}`)) }
process.exit(fail > 0 ? 1 : 0)

File: test-accesstoken-pkg/py/test_docs_api_key.py
"""Mirrors website/public/guides/api-key.md against `imbrace==1.0.4`
(PyPI) — Access Token auth. With access-token, `auth.get_third_party_token`
should succeed.
"""
from __future__ import annotations
import os, sys, json
from dotenv import load_dotenv
load_dotenv()
from imbrace import ImbraceClient

ACCESS_TOKEN = os.environ.get("IMBRACE_ACCESS_TOKEN"); ORG_ID = os.environ.get("IMBRACE_ORGANIZATION_ID")
GATEWAY = os.environ.get("IMBRACE_GATEWAY_URL", "https://app-gatewayv2.imbrace.co")
if not ACCESS_TOKEN or not ORG_ID:
    print("Missing creds"); sys.exit(1)

passed = failed = skipped_n = 0; failures: list[str] = []


def step(label, fn):
    global passed, failed
    sys.stdout.write(f"  - {label} ... "); sys.stdout.flush()
    try:
        r = fn(); print(f"OK {json.dumps(r, default=str)[:90]}"); passed += 1
    except Exception as e:
        print(f"FAIL [{str(e)[:80]}]"); failed += 1; failures.append(f"{label} -> {str(e)[:80]}")


def skip(label, reason):
    global skipped_n
    print(f"  - {label}  SKIP: {reason}"); skipped_n += 1


print(f"\n=== DOCS: guides/api-key.md - auth: ACCESS TOKEN (PyPI imbrace==1.0.4) ===\n")

print("== §Option 2 — programmatic via SDK (access-token expected to succeed) ==")
client = ImbraceClient(access_token=ACCESS_TOKEN, organization_id=ORG_ID, gateway=GATEWAY, timeout=10)
step("client.auth.get_third_party_token(expiration_days=30)",
     lambda: client.auth.get_third_party_token(expiration_days=30))

print("== §Using the key — covered in test-api-pkg ==")
skip("ImbraceClient(api_key=...).platform.get_me()", "covered in test-api-pkg")

print(f"\n=== Summary (api-key / Access Token) ===")
print(f"pass={passed}  fail={failed}  skip={skipped_n}")
if failures:
    print("Failures:")
    for f in failures: print(f"  - {f}")
sys.exit(1 if failed > 0 else 0)
